### ISSUE TITLE
reversed_empty_ranges should suggest alternatives for desirably empty ranges

### ISSUE THREAD

**Issue by djc** [rust-lang/rust-clippy#5628]
With clippy 1.45.0-nightly (0aa6751c1 2020-05-20), I'm getting the following lint:

```rust
error: this range is empty and using it to index a slice will always yield an empty slice
   --> askama_shared/src/parser.rs:143:16
    |
143 |         (s, &s[0..0], &s[0..0])
    |                ^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#reversed_empty_ranges
```

As ought to be obvious to the reader, these slices are empty on purpose. However, neither the lint itself nor the further information explains a good alternative for empty slices (and some googling also didn't turn up anything so far). In particular, `&[]` doesn't work.

**Comment by ebroto** [comment]
Hey @djc, TBH I thought the use cases for actually wanting an empty slice would be rare and that it would be more often a developer error (the lint works with constants so it's not necessarily obvious at first sight). So in this case I think an `allow` could make sense.

Otherwise the lint could suggest using `&[] as &[T]` which does not require a binding. Would that work for you?

A third option would be not linting in this case, as it's the less problematic of all the possible problems that the lint tries to warn against. I would go for this if the use case happens to be more common that we thought.

I will be glad to work on making the changes whatever the outcome is :)

**Comment by djc** [comment]
Yeah, I feel like the case of two identical literals should probably just be allowed. Adding the suggestion about casting using `as` sounds like a good idea either way!
